sdcm/sct_runner.py

```diff
@@ -797,19 +798,27 @@ def _create_instance(self,  # pylint: disable=too-many-arguments
                 "admin_username": self.LOGIN_USER,
                 "admin_public_key": self.key_pair.public_key.decode(),
             }
+            return azure_region.create_virtual_machine(
+                vm_name=instance_name,
+                vm_size=instance_type,
+                image=base_image,
+                disk_size=self.instance_root_disk_size(test_duration=test_duration),
+                tags=tags | {"launch_time": get_current_datetime_formatted()},
+                **additional_kwargs,
+            )
         else:
-            azure_region = self.azure_region
-            additional_kwargs = {
-                "os_state": AzureOsState.SPECIALIZED,
-            }
-        return azure_region.create_virtual_machine(
-            vm_name=instance_name,
-            vm_size=instance_type,
-            image=base_image,
-            disk_size=self.instance_root_disk_size(test_duration=test_duration),
-            tags=tags | {"launch_time": get_current_datetime_formatted()},
-            **additional_kwargs,
-        )
+            test_id = tags["TestId"]
+            provisioner = provisioner_factory.create_provisioner(backend="azure", test_id=test_id,
+                                                                 region=self.azure_region.location)
+            vm_params = InstanceDefinition(name=instance_name,
+                                           image_id=base_image["id"],
+                                           type=instance_type,
+                                           user_name=None,
+                                           ssh_public_key=None,
```

Comment:
It's OK to have no a  public key here?

Reply:
For sct runner we use predefined (specialized) image - which already has ssh keys imported (also user prepared).